File: lv1-library-loan/csharp/LibraryLoan.Application/Members/InMemoryMemberRepository.cs
using LibraryLoan.Domain.Members;

namespace LibraryLoan.Application.Members;

public sealed class InMemoryMemberRepository : IMemberRepository
{
    private readonly Dictionary<MemberId, Member> _members = new();

    public Member? Find(MemberId id) => _members.GetValueOrDefault(id);

    public void Save(Member member) => _members[member.Id] = member;
}

File: lv1-library-loan/typescript/src/application/members/in-memory-member-repository.ts
import type { Member } from "../../domain/members/member.js";
import type { MemberId } from "../../domain/members/member-id.js";
import type { MemberRepository } from "./member-repository.js";

export class InMemoryMemberRepository implements MemberRepository {
  private readonly members = new Map<string, Member>();

  find(id: MemberId): Member | undefined {
    return this.members.get(id.value);
  }

  save(member: Member): void {
    this.members.set(member.id.value, member);
  }
}
